src/megachatapi_impl.cpp

```diff
@@ -2768,6 +2768,17 @@ const char *MegaChatApiImpl::getUserLastnameFromCache(MegaChatHandle userhandle)
     if (mClient && mClient->isUserAttrCacheReady())
     {
         return move(mClient->userAttrCache().getDataFromCache(userhandle, ::mega::MegaApi::USER_ATTR_LASTNAME)->c_str());
+const char *MegaChatApiImpl::getUserFullnameFromCache(MegaChatHandle userhandle)
+{
+    SdkMutexGuard g(sdkMutex);
+    if (mClient && mClient->isUserAttrCacheReady())
+    {
+        const Buffer* buffer = mClient->userAttrCache().getDataFromCache(userhandle, USER_ATTR_FULLNAME);
+        if (buffer != nullptr)
+        {
+            std::string fullname(buffer->buf()+1, buffer->dataSize()-1);
+            return MegaApi::strdup(fullname.c_str());
```

Comment:
Since we give the ownership to the caller, they need the actual pointer returned by `new []`, otherwise it will result on undefined behavior.
In short, disregard my suggestion.